langmeta/langmeta/shared/src/main/scala/org/langmeta/semanticdb/Symbol.scala

```diff
@@ -33,128 +33,49 @@ object Symbol {
-        if (currChar == EOF) {
-          curr
-        } else if (currChar == ';') {
-          curr
-        } else if (currChar == '[') {
-          readChar()
-          val name = parseName()
-          if (currChar != ']') fail()
-          else readChar()
-          parseGlobal(Symbol.Global(curr, Signature.TypeParameter(name)))
-        } else if (currChar == '(') {
-          readChar()
-          val name = parseName()
-          if (currChar != ')') fail()
-          else readChar()
-          parseGlobal(Symbol.Global(curr, Signature.TermParameter(name)))
-        } else {
-          val name = parseName()
-          if (currChar == '#') {
-            readChar()
-            parseGlobal(Symbol.Global(curr, Signature.Type(name)))
-          } else if (currChar == '.') {
-            readChar()
-            parseGlobal(Symbol.Global(curr, Signature.Term(name)))
-          } else if (currChar == '(') {
-            val buf = new StringBuilder()
-            buf += currChar
-            while (readChar() != '.') buf += currChar
-            readChar()
-            parseGlobal(Symbol.Global(curr, Signature.Method(name, buf.toString)))
-          } else if (currChar == '=') {
-            readChar()
-            if (currChar != '>') fail()
-            else readChar()
-            parseGlobal(Symbol.Global(curr, Signature.Self(name)))
-          } else {
-            fail()
-          }
-        }
-      }
-      def parseLocal(): Symbol = {
-        val name = parseName()
-        if (name.startsWith("local")) Symbol.Local(name)
-        else fail()
-      }
-
-      def parseMulti(symbols: List[Symbol]): Symbol = {
-        if (currChar == EOF) {
-          symbols match {
-            case Nil => Symbol.None
-            case List(symbol) => symbol
-            case symbols => Symbol.Multi(symbols.sortBy(_.syntax))
-          }
-        } else {
-          val symbol = {
-            if (currChar == '_') parseGlobal(Symbol.None)
-            else parseLocal()
-          }
-          if (currChar == ';') {
-            readChar()
-            if (currChar == EOF) fail()
-          }
-          parseMulti(symbols :+ symbol)
-        }
-      }
-
-      def entryPoint(): Symbol = {
-        readChar()
-        parseMulti(Nil)
-      }
+  import fastparse.all._
+  private def charsExcept(c: Char) =
+    (!c.toString ~ AnyChar).rep(min = 0).!
+
+  private val quotedName =
+    P("`" ~ charsExcept('`').! ~ "`")
+  private val unquotedName =
+    P(CharPred(Character.isJavaIdentifierStart) ~ CharsWhile(Character.isJavaIdentifierPart, min = 0)).!
+  private val name =
+    P(quotedName | unquotedName)
+
+  private val signature = P {
+    ("[" ~/ name ~ "]").map(Signature.TypeParameter) |
+    ("(" ~/ name ~ ")").map(Signature.TermParameter) |
+    (name ~/ (
+      "#" ~/ PassWith(Signature.Type) |
+      "." ~/ PassWith(Signature.Term) |
+      (("(" ~/ charsExcept('.')).! ~ ".").map(sig => Signature.Method(_: String, sig)) |
```

Comment:
I think you can cut after `charsExcept('.')).!`

Reply:
fine